website/src/components/Chat/ChatConversation.tsx

```diff
@@ -108,25 +117,113 @@ export const ChatConversation = memo(function ChatConversation({ chatId, getConf
       }
       if (message) {
         setMessages((messages) => [...messages, message!]);
+        setActiveMessageId(message.id);
       }
       setQueueInfo(null);
       setResponse(null);
       setIsSending.off();
       setShowEncourageMessage.on();
     },
-    [getConfigValues, setIsSending, setShowEncourageMessage, toast]
+    [getConfigValues, setIsSending, setShowEncourageMessage, toast, setActiveMessageId]
+  );
+  const createAssistantDrafts = useCallback(
+    async ({ parentId, chatId }: { parentId: string; chatId: string }) => {
+      const { model_config_name, plugins, ...sampling_parameters } = getConfigValues();
+
+      const assistant_arg: InferencePostAssistantMessageParams = {
+        chat_id: chatId,
+        parent_id: parentId,
+        model_config_name: model_config_name,
+        sampling_parameters: sampling_parameters,
+        plugins: plugins,
+      };
+
+      let draft_messages: InferenceMessage[];
+      try {
+        const promises = Array.from({ length: NUM_GENERATED_DRAFTS }, () =>
+          post(API_ROUTES.CREATE_ASSISTANT_MESSAGE, { arg: assistant_arg })
+        );
+
+        draft_messages = await Promise.all(promises);
+      } catch (err) {
+        if (err instanceof OasstError) {
+          toast({
+            title: err.message,
+            status: "error",
+          });
+        }
+        setIsSending.off();
+        return;
+      }
+
+      setStreamedDrafts(Array(NUM_GENERATED_DRAFTS).fill(""));
+      const complete_draft_messages = await Promise.all(
+        draft_messages.map(async (draft_message, index) => {
+          const { body, status } = await fetch(API_ROUTES.STREAM_CHAT_MESSAGE(chatId, draft_message.id));
+
+          let inference_message: InferenceMessage | null;
+          if (status === 204) {
+            inference_message = await get(API_ROUTES.GET_MESSAGE(chatId, draft_message.id));
+          } else {
+            inference_message = await handleChatEventStream({
+              stream: body!,
+              onError: console.error,
+              onPending: setQueueInfo,
```

Comment:
This line causes a lot of race condition, and make the queue info flashing all the times. Can we call setQueueInfo only 1 time or move the queue info to inside the message?